src/Microsoft.DocAsCode.Build.Engine/PrebuildBuildPhaseHandlerWithIncremental.cs

```diff
@@ -95,9 +95,78 @@ private void PostHandle(List<HostService> hostServices)
             {
                 h.SaveIntermediateModel(IncrementalContext);
             }
+            ReportDependency(hostServices);
             Logger.UnregisterListener(CurrentBuildMessageInfo.GetListener());
         }
 
+        private void ReportDependency(IEnumerable<HostService> hostServices)
+        {
+            ReportFileDependency(hostServices);
+            ReportUidDependency(hostServices);
+        }
+
+        private void ReportUidDependency(IEnumerable<HostService> hostServices)
+        {
+            foreach (var hostService in hostServices)
+            {
+                foreach (var m in hostService.Models)
+                {
+                    if (m.Type == DocumentType.Overwrite)
+                    {
+                        continue;
+                    }
+                    if (m.LinkToUids.Count != 0)
+                    {
+                        string fromNode = ((RelativePath)m.OriginalFileAndType.File).GetPathFromWorkingFolder().ToString();
+                        var dps = from f in GetFilesFromUids(m.LinkToUids)
+                                  select new DependencyItem(fromNode, f, fromNode, DependencyTypeName.Uid);
+                        CurrentBuildVersionInfo.Dependency.ReportDependency(dps);
+                    }
+                }
+            }
+        }
+
+        private void ReportFileDependency(IEnumerable<HostService> hostServices)
+        {
+            foreach (var hostService in hostServices)
+            {
+                foreach (var m in hostService.Models)
+                {
+                    if (m.Type == DocumentType.Overwrite)
```

Comment:
i would add the dependency in case later we decide to support incremental for  managedreferenceprocessor

---
In reply to: [89744162](https://github.com/dotnet/docfx/pull/932#discussion_r89744162) [](ancestors = 89744162,89735773)